packages/ifpack2/src/Ifpack2_BlockTriDiContainer_decl.hpp

```diff
@@ -0,0 +1,362 @@
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Michael A. Heroux (maherou@sandia.gov)
+//
+// ***********************************************************************
+//@HEADER
+*/
+
+#ifndef IFPACK2_BLOCKTRIDICONTAINER_DECL_HPP
+#define IFPACK2_BLOCKTRIDICONTAINER_DECL_HPP
+
+/// \file Ifpack2_BlockTriDiContainer_decl.hpp
+/// \brief Ifpack2::BlockTriDiContainer class declaration
+
+#include "Ifpack2_Container.hpp"
+#include "Tpetra_MultiVector.hpp"
+#include "Tpetra_Map.hpp"
+#include "Tpetra_RowMatrix.hpp"
+#include "Tpetra_BlockCrsMatrix_decl.hpp"
+#include <type_traits>
+#include <string>
+
+namespace Ifpack2 {
+
+namespace Details { namespace Todo { int test_add_radial(); }}
+
+// Impl for BlockTriDiContainer. Can't be an internal class because we need to
+// partially specialize Impl.
+namespace BlockTriDiContainerDetails {
+template <typename MatrixType, typename LocalScalarType, typename ExeSpace> class Impl;
+}
+
+/// \class BlockTriDiContainer
+/// \brief Store and solve local block tridiagonal linear problems.
+/// \tparam MatrixType A specialization of Tpetra::RowMatrix.
+///
+/// This class can be used as a Container for BlockRelaxation, in which case
+/// please refer to the documentation of the Container interface, or
+/// standalone. In standalone use, there are special constructor,
+/// <tt>compute</tt>, and <tt>applyInverseJacobi</tt> functions that may be
+/// called, with extra non-ParameterList inputs.
+///
+/// BlockTriDiContainer requires MatrixType to be Tpetra::BlockCrsMatrix. It
```

Comment:
With Ifpack2, MatrixType should always be RowMatrix, right?  This should be a subclass of BlockCrsMatrix.